spec/new_integration/web_app_spec.rb

```diff
@@ -20,13 +20,17 @@ class Application < Hanami::Application
       RUBY
 
       write "config/routes.rb", <<~RUBY
-        Hanami.application.routes do
-          slice :main, at: "/" do
-            root to: "home#index"
-          end
+        module TestApp
+          class Routes < Hanami::Application::Routes
+            define do
+              slice :main, at: "/" do
+                root to: "home#index"
+              end
 
-          slice :admin, at: "/admin" do
-            get "/dashboard", to: "dashboard#show"
+              slice :admin, at: "/admin" do
+                get "/dashboard", to: "dashboard#show"
```

Comment:
@waiting-for-dev How this could work if the new separator is a `.` instead of the old `#`? 😅 
Old: `"dashboard#show"`, new `"dashboard.show"`.

Reply:
That's something that hasn't changed with this PR. Here we're using the same routes definition example as before. The only difference is that they're defined in a class. Of course, we must study it, but it's not related to this PR :slightly_smiling_face:

Reply:
The separators `#` and `/` are accepted here (as well as the standard `.`). The built-in application endpoint resolver converts them back to dots as it resolves the object from the container: https://github.com/hanami/hanami/blob/f93c4d2618171c6527cd18d1d394b61ba3f87d9c/lib/hanami/application/routing/resolver.rb#L75

We can probably tweak that a bit, but I did this purposefully to preserve existing behaviour, and I figured the `#` actually helped provide a little more meaning to the endpoint names.

Happy to discuss elsewhere :)